source/common/network/utility.cc

```diff
@@ -272,7 +287,7 @@ bool Utility::isLocalConnection(const Network::ConnectionSocket& socket) {
       const auto local_address = Address::addressFromSockAddr(
           *addr, (af_look_up == AF_INET) ? sizeof(sockaddr_in) : sizeof(sockaddr_in6));
 
-      if (remote_address == local_address) {
+      if (*remote_address == *local_address) {
```

Comment:
This comparison from before is also wrong I believe, as Address::addressFromSockAddr() returns a SharedPtr, so this is a pointer comparison that is never going to be true.

I couldn't get the isLocalAddress tests to fail with or without this change, I am not sure this code is well-exercised (and that's probably hard without a mock for getifaddrs).

Which raises a question for me: should we remove the getifaddrs() code entirely?  Unless I'm confused (totally possible :-) ) I don't think anyone ever got a return of true out of this block before.  Seems like calling getifaddrs for each isLocalAddress() call could be expensive.  And checking if the local and remote IP addresses match on a connected socket you created meets my definition of "local".

Reply:
Yeah from looking at the coverage report, this code has no coverage and it is clearly broken as you point out (nice find): https://storage.googleapis.com/envoy-coverage/report-master/coverage/proc/self/cwd/source/common/network/utility.cc.html

Given that it never worked and we have no coverage, I would be inclined to delete all of it until someone wants to bring it back. @envoyproxy/maintainers any thoughts here?

Reply:
OK since no one else responded I will make the call to kill this broken code which no one must be using. Let's delete it! Thank you.